crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -80,6 +87,51 @@ impl PgIndexerStore {
             module_cache,
         }
     }
+
+    pub async fn get_sui_types_object(
+        &self,
+        object_id: &ObjectID,
+        version: &SequenceNumber,
+    ) -> Result<sui_types::object::Object, IndexerError> {
+        let pg_object = read_only!(&self.cp, |conn| {
+            objects_history::dsl::objects_history
+                .filter(objects_history::object_id.eq(object_id.to_string()))
+                .filter(objects_history::version.eq(version.value() as i64))
+                .first::<Object>(conn)
+        })
+        .context("Failed reading Object from PostgresDB");
+        match pg_object {
+            Ok(pg_object) => {
+                let object = sui_types::object::Object::try_from(pg_object)?;
+                Ok(object)
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    pub async fn find_sui_types_object_lt_or_eq_version(
+        &self,
+        id: &ObjectID,
+        version: &SequenceNumber,
+    ) -> Result<Option<sui_types::object::Object>, IndexerError> {
+        let pg_object = read_only!(&self.cp, |conn| {
+            objects_history::dsl::objects_history
+                .filter(objects_history::object_id.eq(id.to_string()))
+                .filter(objects_history::version.le(version.value() as i64))
```

Comment:
what are your thoughts on lifting
```
                .filter(objects_history::object_id.eq(object_id.to_string()))
                .filter(objects_history::version.eq(version.value() as i64))
``` into its own function

Reply:
we will have some query builder to compose all kind of query fragments, here 2 lines seem fragmented to do it separately.